File: test/fixtures/apps/rpc-server/app/rpc/FooService.js
'use strict';

/**
 * 获取用户信息
 * @param {String} login - 用户域账号
 * @return {Object} 完整用户信息
 * @rpc
 */
exports.getUser = function*(login) {
  return {
    login,
    name: 'foo user',
  };
};

/**
 * [*testType description]
 * @param {int} i - [description]
 * @param {integer} it - [description]
 * @param {number} n - [description]
 * @param {boolean} bl - [description]
 * @param {bool} b - [description]
 * @param {long} l - [description]
 * @param {string} s - [description]
 * @param {object} o - [description]
 * @param {map} map - [description]
 * @param {array} array - [description]
 * @param {list} list - [description]
 * @return {String} [description]
 * @rpc
 */
exports.testType = function*(i, it, n, bl, b, l, s, o, map, array, list) {
  return 'ok';
};

File: test/fixtures/expect/rpc-server/src/main/java/com/eggjs/FooService.java
package com.eggjs;

import java.util.*;

public interface FooService {
  /**
   * 获取用户信息
   * @param login 用户域账号
   * @return 完整用户信息
   */
  public Map getUser(String login);
  
  /**
   * [*testType description]
   * @param i [description]
   * @param it [description]
   * @param n [description]
   * @param bl [description]
   * @param b [description]
   * @param l [description]
   * @param s [description]
   * @param o [description]
   * @param map [description]
   * @param array [description]
   * @param list [description]
   * @return [description]
   */
  public String testType(Integer i, Integer it, double n, Boolean bl, Boolean b, Long l, String s, Map o, Map map, List array, List list);
  
}
